airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/check_stream.py

```diff
@@ -41,6 +41,7 @@ def check_connection(self, source: Source, logger: logging.Logger, config: Mappi
             try:
                 stream_is_available, reason = availability_strategy.check_availability(stream, logger, source)
                 if stream_is_available:
+                    # why do we return here and not try next stream?
```

Comment:
is it a bug or intentional behavior?

Reply:
This looks like a bug to me! Can you add a test which has 2 streams, for which the first is available and the second is not?